pkg/dns/vips_allocator.go

```diff
@@ -66,89 +67,120 @@ func (d *vipsAllocator) Start(stop <-chan struct{}) error {
 	}
 }
 
-func (d *vipsAllocator) synchronize() error {
+func (d *vipsAllocator) createOrUpdateVIPConfigs() error {
 	meshes := core_mesh.MeshResourceList{}
 	err := d.rm.List(context.Background(), &meshes)
 	if err != nil {
 		return err
 	}
 
-	serviceMap := make(map[string]bool)
 	for _, mesh := range meshes.Items {
-		dataplanes := core_mesh.DataplaneResourceList{}
-
-		err := d.rm.List(context.Background(), &dataplanes, store.ListByMesh(mesh.Meta.GetName()))
-		if err != nil {
-			return err
+		if err := CreateOrUpdateVIPConfig(d.persistence, d.rm, d.resolver, d.ipam, mesh.GetMeta().GetName()); err != nil {
+			vipsAllocatorLog.Error(err, "unable to create or update VIP config", "mesh", mesh.GetMeta().GetName())
 		}
+	}
+	return nil
+}
 
-		// TODO: Do we need to reflect somehow the fact this service belongs to a particular `mesh`
-		for _, dp := range dataplanes.Items {
-			if dp.Spec.IsIngress() {
-				for _, service := range dp.Spec.Networking.Ingress.AvailableServices {
-					serviceMap[service.Tags[mesh_proto.ServiceTag]] = true
-				}
-			} else {
-				for _, inbound := range dp.Spec.Networking.Inbound {
-					serviceMap[inbound.GetService()] = true
-				}
-			}
-		}
+func CreateOrUpdateVIPConfig(p *MeshedPersistence, rm manager.ReadOnlyResourceManager, r DNSResolver, ipam IPAM, mesh string) error {
+	serviceSet, err := BuildServiceSet(rm, mesh)
+	if err != nil {
+		return err
+	}
 
-		externalServices := core_mesh.ExternalServiceResourceList{}
-		err = d.rm.List(context.Background(), &externalServices, store.ListByMesh(mesh.Meta.GetName()))
-		if err != nil {
-			return err
-		}
+	global, err := p.Get()
```

Comment:
do we need global at all?

Reply:
Yes, we need it to have the same VIP for services across Meshes